x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_modal/datafeed_modal.tsx

```diff
@@ -102,40 +114,49 @@ export const DatafeedModal: FC<DatafeedModalProps> = ({ jobId, end, onClose }) =
   const handleChange = (date: moment.Moment) => setEndDate(date);
 
   const handleEndDateChange = (direction: ChartDirectionType) => {
-    if (interval === undefined) return;
+    if (data.bucketSpan === undefined) return;
 
     const newEndDate = endDate.clone();
-    const [count, type] = interval.split(' ');
+    const unitMatch = data.bucketSpan.match(/[d | h| m | s]/g)!;
```

Comment:
Not necessary for this PR but it would be good have some unit testing for what the `newEndDate` is expected to be.